server/src/main/java/org/opensearch/indices/replication/checkpoint/ReplicationCheckpoint.java

```diff
@@ -30,37 +32,59 @@ public class ReplicationCheckpoint implements Writeable, Comparable<ReplicationC
     private final long segmentsGen;
     private final long segmentInfosVersion;
     private final long length;
+    private final String latestSupportedCodec;
```

Comment:
nit: This variabled name seems not appropriate as we are using the codec used in engine.